crates/fuel-streams-core/src/utxos/types.rs

```diff
@@ -0,0 +1,95 @@
+use fuel_core_types::{
+    fuel_tx::{
+        input::message::compute_message_id,
+        Address,
+        Bytes32,
+        MessageId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+};
+// ------------------------------------------------------------------------
+// Utxos
+// ------------------------------------------------------------------------
+pub use fuel_core_types::{
+    fuel_tx::{Receipt, Transaction, UniqueIdentifier},
+    services::txpool::TransactionStatus as FuelCoreTransactionStatus,
+};
+use serde::{Deserialize, Serialize};
+// use super::{types::UtxoType, Address, Bytes32, MessageId, Nonce};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Utxo {
+    pub id: UtxoId,
+    pub sender: Option<Address>,
+    pub recipient: Option<Address>,
+    pub nonce: Option<Nonce>,
+    pub data: Option<Vec<u8>>,
+    pub amount: Option<u64>,
+    pub tx_id: Option<Bytes32>,
```

Comment:
Uxos originate from transactions, so this can't be optional:

```suggestion
    pub tx_id: Bytes32,
```

Reply:
true, fixed